### ISSUE TITLE
Device will be offline when at IDLE mode after about 30min in AVS SDK 3.0.0

### ISSUE THREAD

**Issue by tinypancake** [alexa/avs-device-sdk#2077]
**IMPORTANT**: Before you create an issue, please take a look at our [Issue Reporting Guide](https://github.com/alexa/avs-device-sdk/wiki/Issue-Reporting-Guide).

## Briefly summarize your issue:
After we have integrated AVS Device SDK 3.0.0, when testing connection and normal test, it is shown offline on Alexa App after about 30 mins

## What is the expected behavior?
always online except for powering off

## What behavior are you observing?
Device will repeat the ping http action every 5 minutes, but device offline on Alexa App
```
2023-02-23 04:03:31.306 [   46] 5 HTTP2Transport:sendMessagesAndPings:this=0x44e798:CreatingPingHandler
2023-02-23 04:03:31.306 [   46] 5 PingHandler:create:context=0x44e7e8
2023-02-23 04:03:31.306 [   46] 5 ExchangeHandler:init:context=0x44e7e8
2023-02-23 04:03:31.306 [   46] 5 PingHandler:init:context=0x44e7e8
2023-02-23 04:03:31.306 [   46] 7 HTTP2Transport:createAndSendRequest:this=0x44e798,type=GET
2023-02-23 04:03:31.307 [   46] 5 PingHandler:getRequestHeaderLines
2023-02-23 04:03:31.307 [   46] 5 ConditionVariableWrapper:waitInner
2023-02-23 04:03:31.307 [   45] 9 LibcurlHTTP2Connection:insertActiveStream:this=0x40be38,handle=0xada02c88,streamId=AVSPing-17
2023-02-23 04:03:31.860 [   45] 9 LibcurlHTTP2Request:headerCallback:id=AVSPing-17,size=1,nmemb=13,userData=0xada25f98
2023-02-23 04:03:31.860 [   45] 5 PingHandler:onReceiveResponseCode:responseCode=204
2023-02-23 04:03:31.860 [   45] 9 HTTP2Transport:onActivity:this=0x44e798
2023-02-23 04:03:31.860 [   45] 5 PingHandler:reportPingAcknowledged
2023-02-23 04:03:31.860 [   45] 7 HTTP2Transport:onPingRequestAcknowledged:this=0x44e798,success=true
2023-02-23 04:03:31.860 [   45] 7 HTTP2Transport:checkTimeSpan:this=0x44e798,time_span=300,m_countOfUnexpectedPingRequests=1
2023-02-23 04:03:31.860 [   45] 5 ConditionVariableWrapper:notifyAll
2023-02-23 04:03:31.860 [   45] 5 PingHandler:onReceiveHeaderLine:line=HTTP/2 204 
2023-02-23 04:03:31.860 [   45] 9 HTTP2Transport:onActivity:this=0x44e798
2023-02-23 04:03:31.860 [   46] 9 ConditionVariableWrapper:waitForInner:duration=299999742981
2023-02-23 04:03:31.861 [   45] 9 LibcurlHTTP2Request:headerCallback:id=AVSPing-17,size=1,nmemb=56,userData=0xada25f98
2023-02-23 04:03:31.861 [   45] 5 PingHandler:onReceiveHeaderLine:line=x-amzn-requestid\: 75e87aad-7a0e-42fe-b22f-2cf5fc4af4b2
2023-02-23 04:03:31.861 [   45] 9 HTTP2Transport:onActivity:this=0x44e798
2023-02-23 04:03:31.861 [   45] 9 LibcurlHTTP2Request:headerCallback:id=AVSPing-17,size=1,nmemb=32,userData=0xada25f98
2023-02-23 04:03:31.861 [   45] 5 PingHandler:onReceiveHeaderLine:line=access-control-allow-origin\: *
2023-02-23 04:03:31.861 [   45] 9 HTTP2Transport:onActivity:this=0x44e798
2023-02-23 04:03:31.861 [   45] 9 LibcurlHTTP2Request:headerCallback:id=AVSPing-17,size=1,nmemb=2,userData=0xada25f98
2023-02-23 04:03:31.861 [   45] 5 PingHandler:onReceiveHeaderLine:line=
2023-02-23 04:03:31.861 [   45] 9 HTTP2Transport:onActivity:this=0x44e798
2023-02-23 04:03:31.861 [   45] 5 PingHandler:onResponseFinished:status=COMPLETE
2023-02-23 04:03:31.861 [   45] 5 PingHandler:reportPingAcknowledged
2023-02-23 04:03:31.861 [   45] 7 LibcurlHTTP2Connection:streamFinished:this=0x40be38,streamId=AVSPing-17,result=No error,CURLcode=0
2023-02-23 04:03:31.861 [   45] 9 LibcurlHTTP2Connection:releaseStream:this=0x40be38,streamId=AVSPing-17
2023-02-23 04:03:31.861 [   45] 5 PingHandler:destroy
```
## Provide the steps to reproduce the issue, if applicable:
(1) Power on Device
(2) Do nothing, just put device on desk with network connected
(3) Device offline after 30 minutes

## Tell us about your environment:
```
sh-5.1# curl --version
curl 7.82.0 (arm-poky-linux-gnueabi) libcurl/7.82.0 OpenSSL/3.0.7 nghttp2/1.47.0
Release-Date: 2022-03-05
Protocols: file ftp ftps http https
Features: alt-svc Debug HSTS HTTP2 Largefile NTLM SSL TLS-SRP UnixSockets
```

### What version of the AVS Device SDK are you using?
      <3.0.0>

### Tell us what hardware you're using:
- [ ] Other - tell us more:

### Tell us about your OS (Type & version):
- [ ] Linux

**Comment by tinypancake** [comment]
Here is the offline log with level DEBUG9

[avs-offline-issue.log](https://github.com/alexa/avs-device-sdk/files/10813648/avs-offline-issue.log)

**Comment by tinypancake** [comment]
I have found that no SERVER_SIDE_DISCONNECT occurs every 50~55mins, maybe it relate to the SDK Arch or internal flow change?

**Comment by tinypancake** [comment]
I have resolved this issue with curl 7.88.1
